py_mmd_tools/nc_to_mmd.py

```diff
@@ -0,0 +1,328 @@
+                if 'OGC WMS' in prot_type:
+                    wms_layers = ET.SubElement(dacc, ET.QName(ns_map['mmd'], 'wms_layers'))
+                    # Don't add variables containing these names to the wms layers
+                    skip_layers = ['latitude', 'longitude', 'angle']
+                    for w_layer in all_netcdf_variables:
+                        if any(skip_layer in w_layer for skip_layer in skip_layers):
+                            continue
+                        wms_layer = ET.SubElement(wms_layers, ET.QName(ns_map['mmd'], 'wms_layer'))
+                        wms_layer.text = w_layer
+                    # Need to add get capabilities to the wms resource
+                    res += '?service=WMS&version=1.3.0&request=GetCapabilities'
+                dacc_res.text = res
+
+        # Add OGC WMS data_access as comment
+        root.append(ET.Comment(str('<mmd:data_access>\n\t<mmd:type>OGC WMS</mmd:type>\n\t<mmd:description>OGC Web '
+                                   'Mapping Service, URI to GetCapabilities Document.</mmd:description>\n\t'
+                                   '<mmd:resource></mmd:resource>\n\t<mmd:wms_layers>\n\t\t<mmd:wms_layer>'
+                                   '</mmd:wms_layer>\n\t</mmd:wms_layers>\n</mmd:data_access>')))
+
+        et = ET.ElementTree(ET.fromstring(ET.tostring(root, pretty_print=True).decode("utf-8")))
+        et.write(str(self.output_file), pretty_print=True)
+
+    def required_mmd_elements(self):
+        """ Create dict with required MMD elements"""
+
+        mmd_required_elements = {
+                'metadata_identifier': None,
+                'metadata_status': None,
+                'collection': None,
+                'title': None,
+                'abstract': None,
+                'last_metadata_update': None,
+                'dataset_production_status': None,
+                'operational_status': None,
+                'iso_topic_category': None,
+                'keywords': '\n\t\t<mmd:keyword></mmd:keyword>\n\t',
+                'temporal_extent': '\n\t\t<mmd:start_date></mmd:start_date>\n\t',
+                'geographic_extent': str('\n\t\t<mmd:rectangle srsName=""> \n\t\t\t<mmd:north></mmd:north> \n\t\t\t'
+                                         '<mmd:south></mmd:south> \n\t\t\t<mmd:east></mmd:east> \n\t\t\t<mmd:west>'
+                                         '</mmd:west> \n\t\t</mmd:rectangle>\n\t')
+            }
+        return mmd_required_elements
+
+    def generate_cf_mmd_lut_missing_acdd(self):
+        """ Translation dict between ACDD and MMD (dict={ACDD: MDD})
+
+        Create lookup table for mandatory MMD elements missing in the ACDD
+        but that still is present as global attributes in the netCDF file"""
+
+        cf_mmd = {'metadata_status': 'metadata_status',
+                  'collection': 'collection',
+                  'dataset_production_status': 'dataset_production_status',
```

Comment:
Originating back to the implementation we used in Damocles. This will be part of our request to ACDD governance. Concerning structure, see my comment above.

Reply:
I would not call this MMD requirements, but general requirements. We had this during Damocles many years ago, before we knew of ACDD...